tpot/builtins/embedding_estimator.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""This file is part of the TPOT library.
+
+TPOT was primarily developed at the University of Pennsylvania by:
+    - Randal S. Olson (rso@randalolson.com)
+    - Weixuan Fu (weixuanf@upenn.edu)
+    - Daniel Angell (dpa34@drexel.edu)
+    - and many more generous open source contributors
+
+TPOT is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+TPOT is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with TPOT. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.neural_network import MLPClassifier, MLPRegressor
+
+
+class EmbeddingEstimator(TransformerMixin, BaseEstimator):
+    """Meta-transformer for creating neural network embeddings as features.
+    """
+
+    def __init__(self, estimator, embedding_layer=None, backend=None):
+        """Create a StackingEstimator object.
+
+        Parameters
+        ----------
+        estimator: neural network model; either from sklearn or Keras-like.
+            The estimator to generate embeddings.
+        embedding_layer: the particular layer used as the embedding. 
+            By default we use the second last layer. Layers are counted with
+            input layer being `0th` layer; negative indices are allowed. 
+        backend: (optional), the backend we use to query the neural network. 
+            Not required if using scikit-learn interface.
+            Currently only supports keras-like interface (incl. tensorflow)
+        """
+        second_last_layer = -2
+        self.estimator = estimator
+        self.embedding_layer = (
+            second_last_layer if embedding_layer is None else embedding_layer
+        )
+        self.backend = backend
+
+    def fit(self, X, y=None, **fit_params):
+        """Fit the StackingEstimator meta-transformer.
+
+        Parameters
+        ----------
+        X: array-like of shape (n_samples, n_features)
+            The training input samples.
+        y: array-like, shape (n_samples,)
+            The target values (integers that correspond to classes in classification, real numbers in regression).
+        fit_params:
+            Other estimator-specific parameters.
+
+        Returns
+        -------
+        self: object
+            Returns a copy of the estimator
+        """
+        if not issubclass(self.estimator.__class__, MLPClassifier) and not issubclass(
```

Comment:
- not sure if this is the best way to check for Keras etc.

Q: how would we support pure tensorflow or pytorch? I've used Keras here to try to use the Keras scikit-learn wrapper

Reply:
Thank you for this PR. Keras scikit-learn wrapper is good so far.